devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java

```diff
@@ -262,13 +265,11 @@ private AppArtifact initAppArtifact(QuarkusBootstrapMojo mojo) throws MojoExecut
         }
 
         @Override
-        public void close() throws IOException {
-            if (curatedApp != null) {
-                curatedApp.close();
-                curatedApp = null;
-            }
-            this.artifactResolver = null;
-            this.quarkusBootstrap = null;
+        public void close() {
+            // TODO It seems this close method is never call.
```

Comment:
It should be. Do you have `<extensions>true</extensions>` in the `quarkus-maven-plugin` config?

Reply:
Good catch! I was not aware of this configuration. It might make the modifications in `QuarkusBootstrapProvider` useless (I will check).

Reply:
So the `close()` is indeed properly called. I will remove the comment.
We still have to do the change in the `QuarkusBootstrapProvider`[1] though, as the `close()` is called only once at the end of the maven execution.

[1] (the change)[https://github.com/quarkusio/quarkus/pull/16054/files#diff-a3d8d753534cdccadf2cf2437e8d090dd79e67832603e99486696369e8328f62R39] allows to store multiple instances of `QuarkusAppBootstrapProvider`, each of them containing specific properties defined in each `<execution>` of the plugin.